### PR TITLE
Fixes 438

### PR DESCRIPTION
**Pull request by mmahalwy**
- Fixes 438
- Fixes the hot module reloading that @sharabash missed
- Adds server compilation via babel, this needs to be tested @ahmedre 

This should inshallah squeeze some speed on the server and client side.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,9 +14,11 @@
     "posttest:dev:functional": "bash tests/functional/test.sh stop",
     "test:dev:lint": "node_modules/eslint/bin/eslint.js ./src/scripts/**/*.js",
     "test:stylelint": "node_modules/stylelint/dist/cli.js './src/**/*.scss' --config webpack/.stylelintrc",
-    "dev": "node webpack/dev-server.js & PORT=8000 node start.js",
+    "dev": "env NODE_PATH='./src' PORT=8000 UV_THREADPOOL_SIZE=100 node webpack/webpack-dev-server.js & PORT=8000 node ./bin/server.js",
     "start": "NODE_PATH=\"./src\" node ./start",
-    "build": "node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js",
+    "build": "npm run build:client & npm run build:server",
+    "build:server": "babel src -d dist -D",
```
Comment (sabeurthabti) [inline package.json]: you might want to include `--copy-files` to copy over any none js files. 

http://stackoverflow.com/questions/32642685/babel-cli-copy-nonjs-files

Comment (mmahalwy) [inline package.json]: `-D` does that :)

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32773

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32774

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32775

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32776

**Comment by mmahalwy** [comment]
@ahmedre seems i need to do a kick commit for the changes to work...

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32777

**Comment by ahmedre** [comment]
so two things:
1. you don't need a kick update - you can just add a comment that says "rebuild" and it'll rebuild.
2. yep, `Dockerfile` changes are ignored by default for security purposes. i'll test manually insha'Allah and update.

**Comment by ahmedre** [comment]
fails with node 6.3:

```
npm info lifecycle quran@1.0.0~build:client: Failed to exec build:client script
npm ERR! Linux 4.4.0-34-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build:client"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! quran@1.0.0 build:client: `node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the quran@1.0.0 build:client script 'node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the quran package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs quran
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls quran
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /quran/npm-debug.log
```

and with node 5.10.0:

```
npm info lifecycle quran@1.0.0~build:client: Failed to exec build:client script
npm ERR! Linux 4.4.0-34-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build:client"
npm ERR! node v5.10.0
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE
npm ERR! quran@1.0.0 build:client: `node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the quran@1.0.0 build:client script 'node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the quran package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./node_modules/webpack/bin/webpack.js --config webpack/prod.config.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs quran
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls quran
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /quran/npm-debug.log
```

will update with more information in a bit insha'Allah.

File: webpack/prod.config.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ module.exports = {
     sourceMapFilename: '[name]-[chunkhash].map.js'
 
   },
+  context: path.resolve(__dirname, '../src'),
```
Comment (ahmedre) [inline webpack/prod.config.js]: change `../src` to `..`

Comment (ahmedre) [inline webpack/prod.config.js]: line 30 - change `./client.js` to `./src/client.js`

**Comment by ahmedre** [comment]
after the above changes made, the image builds, but unfortunately doesn't work - the server continues, in an infinite loop, to do:

```
05:21:43 server.prod-0 [webpack-isomorphic-tools] (waiting for the first Webpack build to finish)
05:21:43 server.prod-1 [webpack-isomorphic-tools] (waiting for the first Webpack build to finish)
```

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32781

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32782

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32783

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32784

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32787

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32789

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32790

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32792

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32795

**Pull request merged**